File: facebook-common/src/main/java/com/facebook/share/internal/ResultProcessor.java
/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * You are hereby granted a non-exclusive, worldwide, royalty-free license to use,
 * copy, modify, and distribute this software in source code or binary form for use
 * in connection with the web services and APIs provided by Facebook.
 *
 * As with any software that integrates with the Facebook platform, your use of
 * this software is subject to the Facebook Developer Principles and Policies
 * [http://developers.facebook.com/policy/]. This copyright notice shall be
 * included in all copies or substantial portions of the software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
 * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
 * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
 * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
 * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

package com.facebook.share.internal;

import android.os.Bundle;

import com.facebook.FacebookCallback;
import com.facebook.FacebookException;
import com.facebook.internal.AppCall;

/**
 * com.facebook.share.internal is solely for the use of other packages within the
 * Facebook SDK for Android. Use of any of the classes in this package is
 * unsupported, and they may be modified or removed without warning at any time.
 *
 * Callback class to allow derivations of FacebookDialogBase to do custom operations
 */
public abstract class ResultProcessor {
    private FacebookCallback appCallback;

    public ResultProcessor(FacebookCallback callback) {
        this.appCallback = callback;
    }

    public abstract void onSuccess(AppCall appCall, Bundle results);

    /**
     * Override this if anything needs to be done on cancellation (e.g. Logging)
     */
    public void onCancel(AppCall appCall) {
        if (appCallback != null) {
            appCallback.onCancel();
        }
    }

    /**
     * Override this if anything needs to be done on error (e.g. Logging)
     */
    public void onError(AppCall appCall, FacebookException error) {
        if (appCallback != null) {
            appCallback.onError(error);
        }
    }
}

File: facebook-common/src/main/java/com/facebook/share/internal/ResultProcessor.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.share.internal

import android.os.Bundle
import com.facebook.FacebookCallback
import com.facebook.FacebookException
import com.facebook.internal.AppCall

/**
 * com.facebook.share.internal is solely for the use of other packages within the Facebook SDK for
 * Android. Use of any of the classes in this package is unsupported, and they may be modified or
 * removed without warning at any time.
 *
 * Callback class to allow derivations of FacebookDialogBase to do custom operations
 */
abstract class ResultProcessor(private val appCallback: FacebookCallback<*>?) {
  abstract fun onSuccess(appCall: AppCall, results: Bundle?)

  /** Override this if anything needs to be done on cancellation (e.g. Logging) */
  open fun onCancel(appCall: AppCall) {
    appCallback?.onCancel()
  }

  /** Override this if anything needs to be done on error (e.g. Logging) */
  open fun onError(appCall: AppCall, error: FacebookException) {
    appCallback?.onError(error)
  }
}
